### PR TITLE
Try to fix second instance and notifications

### PR DESCRIPTION
**Pull request by charlag**

### CODE REVIEW

**Comment by x80486** [comment]
...merge?

**Comment by charlag** [comment]
I did not work, you can download and check

**Comment by x80486** [comment]
Ah! You are right:

```shell
x80486@ubuntu-vm:~/Workshop/tutanota$ flatpak install --user https://dl.flathub.org/build-repo/2771/com.tutanota.Tutanota.flatpakref
**
flatpak:ERROR:app/flatpak-main.c:700:flatpak_run: assertion failed: (success || error)
Aborted (core dumped)
```

Maybe you can try to incorporate those changes on the pull request for the next release...or maybe triggering another build for this one. I've been having some issues trying to build Node.js, but so far I don't see the same for Tutanota.

**Comment by charlag** [comment]
I think Flatpak build is pretty much unusable right now. You can send a pe and it will start a build.

**Comment by x80486** [comment]
bot, build com.tutanota.Tutanota

**Comment by charlag** [comment]
Ooof
```
running: 
    python tools/gyp_node.py --no-parallel -f make-linux
'import sitecustomize' failed; use -v for traceback
gyp: Call to 'python -c "import sys; print sys.byteorder"' returned exit status 0 while in /run/build/nodejs/deps/v8/gypfiles/v8.gyp. while loading dependencies of /run/build/nodejs/node.gyp while trying to load /run/build/nodejs/node.gyp
Error running GYP
]2;flatpak-builder: Cleanup nodejsError: module nodejs: Child process exited with code 1
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64-4/build/.flatpak-builder/rofiles/rofiles-loXMPn
```
what's that

**Comment by x80486** [comment]
Errrr...looks like we are on _storm season_; now you've got into the same state I'm in. I filed an [issue](https://github.com/flathub/flathub/issues/987/) in Flathub for this, but I haven't received any feedback.

**Comment by charlag** [comment]
Thanks, let's see.
Not so reproducible it seems.

**Comment by x80486** [comment]
@charlag, you should be able to move forward with this one by including the Python `2.7` [shared module](https://github.com/flathub/shared-modules/); you can look into the `io.github.liberodark.OpenDrive` manifest for an example.

**Comment by charlag** [comment]
Thanks for letting me know, the question is why the hell we need Python at all

**Comment by x80486** [comment]
> Thanks for letting me know, the question is why the hell we need Python at all

In order to build Node.js you would need Python `2.7.x`. Notice that the same won't be part of the final build – it's just a build dependency that was provided before but not anymore.

At some point when Python `3.x` can be used instead you wouldn't need it anymore :sunglasses:

**Comment by charlag** [comment]
Sorry, it wasn't meant as a literal question. I thought it's only needed for the dependency generator script, didn't know it's needed by node.
Will try it out next week, once the release is done, or you can send a pr

**Comment by charlag** [comment]
Okay, notifications are there but we still don't know wtf is going on with second instance. It seems like each launched instance has its own copy of the files or something like that.

**Comment by charlag** [comment]
I"ve found how single instance lock works:
https://github.com/electron/electron/issues/15304
since we've given permission for `/tmp` already, we need to check what is `TMPDIR` set to inside flatpak sandbox

File: com.tutanota.Tutanota.json [hunk 1]
```diff
@@ -9,18 +9,24 @@
...
     "separate-locales": false,
     "finish-args": [
+        "--persist=.config/tutanota-desktop",
         "--share=ipc",
         "--socket=x11",
         "--socket=pulseaudio",
         "--share=network",
-        "--filesystem=home"
+        "--filesystem=host",
+        "--socket=system-bus",
```
Comment (TingPing) [inline com.tutanota.Tutanota.json]: This is far too broad...

File: com.tutanota.Tutanota.json [hunk 2]
```diff
@@ -9,18 +9,24 @@
     "command": "tutanota-desktop",
     "separate-locales": false,
     "finish-args": [
+        "--persist=.config/tutanota-desktop",
```
Comment (TingPing) [inline com.tutanota.Tutanota.json]: This path does nothing?

Comment (TingPing) [reply com.tutanota.Tutanota.json]: Also `--persist` conflicts with `host` access.

**Comment by TingPing** [comment]
> since we've given permission for /tmp already, we need to check what is TMPDIR set to inside flatpak sandbox

You aren't given that, nor should you.

Add `--env=TMPDIR=/var/tmp`

**Comment by charlag** [comment]
@TingPing thanks for the advice. I've tried multiple options and I've also tried different set of settings locally. I will try out your suggestion. Will `--env=TMPDIR=/var/tmp` give multiple running instances access to the same directory? Because that's what we need for single instance lock.

**Comment by TingPing** [comment]
Yes thats exactly what it does.

**Comment by charlag** [comment]
It does start now but no window is rendered.
Trying to add `--socket=wayland` but I'm open to suggestions. Maybe `--device=dri` messed it up somehow or it still fails to acquire tmp lock

**Comment by charlag** [comment]
bot, build com.tutanota.Tutanota

**Comment by charlag** [comment]
Wow, it works finally.
So, I guess `device=dri` breaks rendering because there is some problem between Flatpak and OpenGL.

**Pull request closed without merging**